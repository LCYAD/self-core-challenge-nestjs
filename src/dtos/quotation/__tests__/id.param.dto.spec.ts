import {
  generateInvalidStringFormatTest,
  generateMissingFieldTest,
  generateValidBaseTest
} from '@tests/helper/dto.helper'

import {
  QuotationIdParamDto,
  quotationIdParamValidExample
} from '../id.param.dto'

describe('QuotationIdParamDto', () => {
  describe('valid response', () => {
    generateValidBaseTest(QuotationIdParamDto, quotationIdParamValidExample)
  })

  describe('invalid response', () => {
    generateMissingFieldTest(
      ['id'],
      QuotationIdParamDto,
      quotationIdParamValidExample
    )
    generateInvalidStringFormatTest(
      [{ key: 'id', invalidString: '1' }],
      QuotationIdParamDto,
      quotationIdParamValidExample
    )
  })
})
