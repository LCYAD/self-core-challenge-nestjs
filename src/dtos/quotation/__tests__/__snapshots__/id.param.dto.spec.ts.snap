// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuotationIdParamDto invalid response for id not a allowed value 1 1`] = `
[
  ValidationError {
    "children": [],
    "constraints": {
      "matches": "id must match /^[0-9]{12}$/ regular expression",
    },
    "property": "id",
    "value": "1",
  },
]
`;

exports[`QuotationIdParamDto invalid response for missing id 1`] = `
[
  ValidationError {
    "children": [],
    "constraints": {
      "isString": "id must be a string",
      "matches": "id must match /^[0-9]{12}$/ regular expression",
    },
    "property": "id",
    "value": undefined,
  },
]
`;
